Source/WebCore/css/parser/CSSParserFastPaths.cpp

```diff
@@ -901,6 +901,45 @@ static RefPtr<CSSValue> parseSimpleTransform(CSSPropertyID propertyID, StringVie
     return parseSimpleTransformList(string.characters16(), string.length());
 }
 
+static RefPtr<CSSValue> parseDisplay(StringView string)
+{
+    ASSERT(!string.isEmpty());
+    auto value = cssValueKeywordID(string);
+
+    switch (value) {
+    // <display-outside>
+    case CSSValueBlock:
+    case CSSValueInline:
+    // <display-inside> (except for CSSValueFlow since it becomes "block")
+    case CSSValueFlex:
+    case CSSValueFlowRoot:
+    case CSSValueGrid:
+    case CSSValueTable:
+    // <display-internal>
+    case CSSValueTableCaption:
+    case CSSValueTableCell:
+    case CSSValueTableColumnGroup:
+    case CSSValueTableColumn:
+    case CSSValueTableHeaderGroup:
+    case CSSValueTableFooterGroup:
+    case CSSValueTableRow:
+    case CSSValueTableRowGroup:
+    // <display-legacy>
+    case CSSValueInlineBlock:
+    case CSSValueInlineFlex:
+    case CSSValueInlineGrid:
+    case CSSValueInlineTable:
+    // Prefixed values
+    case CSSValueWebkitInlineBox:
+    case CSSValueWebkitBox:
+    // No layout support for the full <display-listitem> syntax, so treat it as <display-legacy>
+    case CSSValueListItem:
+        return CSSPrimitiveValue::create(value);
+    default:
+        return nullptr;
```

Comment:
I would handle CSS wide keywords here: `display: inherit / initial / revert / revert-layer`. 

```
if (isCSSWideKeyword(value))
    return CSSPrimitiveValue::create(value);
```

`parseKeywordValue` does this for free, but a special case in `parseDisplay` is probably fine too.

Reply:
Sounds good. Changed.